artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -289,9 +290,23 @@ public void applySetting(final AddressSettings addressSettings) {
          this.pageLimitBytes = null;
       }
 
+      this.pageLimitMessages = addressSettings.getPageLimitMessages();
+      if (firstTime || !Objects.equals(this.pageLimitMessages, originalLimitMessages)) {
+         if (this.cursorProvider != null) {
+            this.cursorProvider.checkClearPageLimit();
+         }
+      }
+
       if (pageLimitBytes != null && pageSize > 0) {
+         Long originalEstimatedMaxPages = this.estimatedMaxPages;
          estimatedMaxPages = pageLimitBytes / pageSize;
          logger.debug("Address {} should not allow more than {} pages", storeName, estimatedMaxPages);
+         if (firstTime || !Objects.equals(estimatedMaxPages, originalEstimatedMaxPages)) {
+            checkNumberOfPages();
+            if (cursorProvider != null) {
+               cursorProvider.checkClearPageLimit();
+            }
+         }
```

Comment:
Similarly, since start() checks at least the page count, and any use of the address checks the message limits, the suggestion was that it not do this for the constructor., but only later if the settings have changed. This instead always does it for the constructor.